### PR TITLE
ISPN-13014 LDAP identity recursive search

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-13014

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/config_examples/server_ldap_realm.xml [hunk 1]
```diff
@@ -16,7 +16,8 @@
                     direct-verification="true">
             <!-- Defines how principals are mapped to LDAP entries. -->
             <identity-mapping rdn-identifier="uid"
-                              search-dn="ou=People,dc=infinispan,dc=org">
+                              search-dn="ou=People,dc=infinispan,dc=org"
+                              search-recursive="false">
```
Comment (wburns) [inline documentation/src/main/asciidoc/topics/config_examples/server_ldap_realm.xml:20]: Should we add documentation for the filter and iterator filter? I, personally, have no clue what that is :)

Comment (tristantarrant) [reply documentation/src/main/asciidoc/topics/config_examples/server_ldap_realm.xml:20]: I've added a comment about `filter-name` but removed `iterator-filter` completely as it makes no sense for our use-case (it is used when retrieving multiple users at once).

**Comment by wburns** [comment]
The PR itself looks good to me, but looks like there is a test failure 
https://ci.infinispan.org/job/Infinispan/job/PR-9296/3/testReport/org.infinispan.server.hotrod.HotRodSniFunctionalTest/testServerWithNotMatchingDefaultAndClientWithSNI/Timeout/

**Comment by wburns** [comment]
Looks like the prior failure has been on master for a while, so integrating this.

**Pull request merged**